Mailin.php

```diff
@@ -0,0 +1,1186 @@
+<?php
+/**
+ * 2007-2014 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/afl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2014 PrestaShop SA
+ * @license   http://opensource.org/licenses/afl-3.0.php Academic Free License (AFL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+class Mailin
+{
+    public $api_key;
+    public $base_url;
+    public $curl_opts = array();
+    public function __construct($base_url, $api_key)
+    {
+        if (!function_exists('curl_init')) {
+            throw new Exception('Mailin requires CURL module');
+        }
+        $this->base_url = $base_url;
+        $this->api_key = $api_key;
+    }
+    /**
+     * Do CURL request with authorization
+     */
+
+    private function doRequest($resource, $method, $input)
+    {
+        $called_url = $this->base_url."/".$resource;
+        $ch = curl_init($called_url);
+        $auth_header = 'api-key:'.$this->api_key;
+        $content_header = "Content-Type:application/json";
+        if (Tools::strtoupper(Tools::substr(PHP_OS, 0, 3)) === 'WIN') {
+            // Windows only over-ride
+            curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+        }
+        curl_setopt($ch, CURLOPT_HTTPHEADER, array($auth_header, $content_header));
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_HEADER, 0);
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $input);
+        $data = curl_exec($ch);
+        if (curl_errno($ch)) {
+            echo 'Curl error: ' . curl_error($ch). '\n';
```

Comment:
You need to manage errors and display them correctly. 
If something wrong happen your client will see something like this (screenshot)

![download 16](https://cloud.githubusercontent.com/assets/6247743/11333772/8da0f740-91d0-11e5-8936-354e43f19913.png)

You can show the error only if the dev mode is enabled for example:

/* Debug only */
if (!defined('_PS_MODE_DEV_')) {
    // your code
}